Add Teleport Team docs (#26523)
gravitational/teleport: docs/pages/choose-an-edition/teleport-team.mdx

---
title: "Introduction to Teleport Team"
description: "Teleport Team makes it quick and easy for small organizations set up secure RBAC for their infrastructure."
tocDepth: 3
---

Teleport Team helps small organizations get started quickly with enabling secure
access to their infrastructure. 

With Teleport Team, the Auth Service and Proxy Service are fully managed, giving
you a scalable, fault tolerant certificate authority that you can use to
configure access from users across the globe to your infrastructure, including
private networks.

After you start a [free trial](https://goteleport.com/signup) of Teleport Team,
you can set up RBAC, enable single sign-on with GitHub or Teleport's own
identity provider, and start protecting resources with Teleport.

```mermaid
flowchart LR
subgraph team["Teleport Team tenant"]
  direction LR
  proxy["Teleport Proxy Service"]
  auth["Teleport Auth Service"]
  proxy<-.->auth
end

> subgraph public["The internet"]

Reviewer: Consider giving this an empty title and no background. "The internet" might be best diagramed as open.

  web["Teleport Web UI"]
  tsh["tsh client"]
end

subgraph private["Private network"]
  service["Teleport Services"]
end 

service-- Reverse tunnels -->proxy
web & tsh--Client traffic-->proxy
```

<Notice type="tip">

For a detailed comparison of Teleport's editions, including how Teleport Team
compares to Teleport Team, see the [Choose an Edition](introduction.mdx) guide.

</Notice>

## Get started with Teleport Team

> This section walks you through the process of registering a local server with a

Reviewer: Unsure about "section" here, since this describes the bulk of the rest of the page.

Teleport Team account. You can then access the server through the browser and
the terminal, and record your session so you can review it later.

### Prerequisites

- A Teleport Team account. You can sign up for a free trial at the following
  page:

  https://goteleport.com/signup/

- Docker installed on your workstation. We will use a Docker container to
  register a server with Teleport via the Teleport SSH Service. You can find
  installation instructions on [Docker's
  website](https://docs.docker.com/get-docker/).

  <Notice type="tip">

  Docker is only required for the local demo environment we use in this guide.
  If you want to get started registering your servers in your infrastructure
  with Teleport, view our documentation on how to [register servers and
  configure access to them](../server-access/introduction.mdx).

  </Notice>

- The `tsh` client tool. Visit our [Installation Guide](../installation.mdx)
  for instructions. If you want to get started quickly, you can skip this for
  now and access Teleport through your browser.

### Step 1/4. Spin up your server

> Spin up a server on your workstation so you can register this server with your
> Teleport Team tenant:

Reviewer: "server" is used redundantly. Consider something like:

```suggestion
Start a docker container on your workstation to register as a server in your
Teleport Team tenant:
```

```code
$ docker run -it ubuntu:22.10 /bin/bash
```

> Your terminal will begin a shell session in the `ubuntu:22.10` container. From
> there, make sure you have `curl` and `telnet` installed. When we run a script to install Teleport later in this guide, the script will require both of these tools.

Reviewer: Line break and `s/./:`.

```suggestion
Your terminal will begin a shell session in the `ubuntu:22.10` container. From
there, make sure you have `curl` and `telnet` installed. When we run a script to
install Teleport later in this guide, the script will require both of these tools:
```

```code
$ apt update && apt install -y curl telnet
```

The rest of the guide assumes that you will run it as a non-`root` user, but the
shell in the container you created earlier is running as root. Alias `sudo` to

> override this:

Reviewer: This sounds odd to me. If it's "the guide" which we wrote, then we could write it to not use sudo. Instead, can we rephrase this to say that the Teleport web UI provides example commands assuming a non-`root` user?

Author: To make this clearer/simpler, I've moved this text just before the instructions to copy the installation script:

````markdown
The Teleport installation script uses `sudo`, which is not installed on your
Docker container. Alias `sudo` in your container terminal to override this:

```code
$ alias sudo=""
```

Copy the script and paste it into the shell you opened earlier on your Docker
container.
````

type](../../img/cloud/getting-started/choose-resource@2x.png)

Select "Ubuntu 14.04+" so you can register your Docker container.

After selecting an OS, you will see a script that you can copy from the Teleport
Web UI:

![Configure resource](../../img/cloud/getting-started/paste-script@2x.png)

Copy the script and paste it into the shell you opened earlier on your Docker
container.

Leave the script running in your shell. Teleport will run in the foreground. You
will see a message in the Web UI that your new server connected successfully:

![Connected
successfully](../../img/cloud/getting-started/successfully-connected@2x.png)

Click **Next**.

Confirm that you will authenticate as the `root` user and click **Next**:

![Choose your user](../../img/cloud/getting-started/set-up-access@2x.png)

From there, you can test the connection and access the server. Click **Start
Session** to connect:

![Start session](../../img/cloud/getting-started/test-connection@2x.png)

<Notice type="tip">

If you get an error connecting to your container, interrupt the `teleport`
process in your container's shell with **Ctrl-C**, then run `teleport start` to
restart the process.

</Notice>

> You should see a terminal prompt in your browser. Your Teleport Team

Reviewer: Suggest specifying that this will happen in a new tab/window.

account routes your SSH connection through the Teleport Proxy Service, which
connects to your container via a reverse tunnel. 

Reverse tunnels like this allow Teleport to manage access to resources like
Kubernetes clusters and remote desktops, even if these resources are running on
private networks.

Execute some commands in your terminal. When you have finished, run `exit` to
end your session.

> In the **Manage Access** view, click **Finish**, then **Browse existing

Reviewer: Related to previous comment, suggest referencing the previous tab/window.
